app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -30,4 +42,85 @@ class MainViewModel(
             }
         }
     }
+
+    fun launchLegacyActivity(): Boolean {
+        return try {
+            val intent = Intent()
+            intent.setClassName(
+                "app.revanced.manager.flutter",
+                "app.revanced.manager.flutter.ExportSettingsActivity"
+            )
+            launcher.launch(intent)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    fun applyLegacySettings(data: String) = viewModelScope.launch {
+        prefs.showAutoUpdatesDialog.update(false)
+        val jsonData = JSONObject(data)
+        val keysIterator = jsonData.keys()
+        while (keysIterator.hasNext()) {
+            try {
+                val key = keysIterator.next()
+                val value = jsonData.get(key)
```

Comment:
`kotlinx.serialization` should be used instead because the rest of the app uses it

Reply:
The app does not use `kotlinx.serialization` at all though.

Reply:
No, it does use the library: https://github.com/ReVanced/revanced-manager/tree/compose-dev/app/src/main/java/app/revanced/manager/network/dto

Reply:
Oh, I am sorry I was mistaking this PR with mine